src/library/dao/playlistdao.cpp

```diff
@@ -974,8 +974,22 @@ void PlaylistDAO::shuffleTracks(const int playlistId, const QList<int>& position
         //qDebug() << "Swapping tracks " << trackAPosition << " and " << trackBPosition;
         trackPositionIds.insert(trackAPosition, trackBId);
         trackPositionIds.insert(trackBPosition, trackAId);
+
+        // TODO: The following use of QList<T>::swap(int, int) is deprecated
+        // and should be replaced with QList<T>::swapItemsAt(int, int)
+        // However, the proposed alternative has just been introduced in Qt
+        // 5.13. Hence, this should be changed as soon as the minimum required
+        // Qt version of Mixx is increased to 5.13. In the meantime, we'll just
+        // disable the warning here.
+        #pragma GCC diagnostic push
+        #pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+        #pragma clang diagnostic push
+        #pragma clang diagnostic ignored "-Wdeprecated-declarations"
         newPositions.swap(newPositions.indexOf(trackAPosition),
```

Comment:
How about guarding the new code for Qt 5.13 behind a QT_VERSION_CHECK and keeping the old code in the #else branch? If the differences are limited to a very close scope this is the preferred way to go.

Reply:
Done.